File: flipt-java/src/main/java/io/flipt/api/models/ListParameters.java
package io.flipt.api.models;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class ListParameters extends CommonParameters {

  @JsonProperty("limit")
  private final Integer limit;

  @JsonProperty("offset")
  private final Integer offset;

  @JsonProperty("pageToken")
  private final String pageToken;

  public ListParameters(Builder builder) {
    super(builder);
    this.limit = builder.limit;
    this.offset = builder.offset;
    this.pageToken = builder.pageToken;
  }

  public Integer getLimit() {
    return limit;
  }

  public Integer getOffset() {
    return offset;
  }

  public String getPageToken() {
    return pageToken;
  }

  public static Builder builder() {
    return new Builder();
  }

  public static class Builder extends CommonParameters.Builder {
    private int limit;
    private int offset;
    private String pageToken;

    private Builder() {}

    public Builder limit(int limit) {
      this.limit = limit;
      return this;
    }

    public Builder offset(int offset) {
      this.offset = offset;
      return this;
    }

    public Builder pageToken(String pageToken) {
      this.pageToken = pageToken;
      return this;
    }

    public ListParameters build() {
      return new ListParameters(this);
    }
  }
}

File: flipt-csharp/src/Models/ListParameters.cs
using System.Text.Json.Serialization;

namespace Flipt.Models;

/// <summary>
/// Parameters for list API requests with pagination.
/// </summary>
public class ListParameters
{
    [JsonPropertyName("limit")]
    public int? Limit { get; set; }

    [JsonPropertyName("offset")]
    public int? Offset { get; set; }

    [JsonPropertyName("pageToken")]
    public string? PageToken { get; set; }

    [JsonPropertyName("reference")]
    public string? Reference { get; set; }
}
